sonar-dotnet-shared-library/src/main/java/org/sonar/plugins/dotnet/tests/DotCoverReportParser.java

```diff
@@ -31,11 +31,15 @@
 
 public class DotCoverReportParser implements CoverageParser {
 
-  private static final Pattern TITLE_PATTERN = Pattern.compile(".*?<title>(.*?)</title>.*", Pattern.DOTALL);
-  private static final Pattern COVERED_LINES_PATTERN_1 = Pattern.compile(
-    ".*<script type=\"text/javascript\">\\s*+highlightRanges\\(\\[(.*?)\\]\\);\\s*+</script>.*",
+  private static final String TITLE_START = "<title>";
+  // the pattern for the information about a sequence point - (lineStart, columnStart, lineEnd, columnEnd, hits)
+  private static final Pattern SEQUENCE_POINT_PATTERN = Pattern.compile("\\[(\\d++),\\d++,(\\d++),\\d++,(\\d++)]");
+  private static final String SEQUENCE_POINTS_GROUP_NAME = "SequencePoints";
+  // the file coverage has a list of sequence points
+  // we use SEQUENCE_POINT_PATTERN below to avoid non-determinism in the regular expression, due to the `[` and `]` characters appearing multiple times
+  private static final Pattern FILE_COVERAGE_PATTERN = Pattern.compile(
+    ".*<script type=\"text/javascript\">\\s*+highlightRanges\\(\\[(?<" + SEQUENCE_POINTS_GROUP_NAME + ">" + SEQUENCE_POINT_PATTERN + "(," + SEQUENCE_POINT_PATTERN + ")*)]\\);\\s*+</script>.*",
```

Comment:
I think the trailing `]` should be escaped
```suggestion
    ".*<script type=\"text/javascript\">\\s*+highlightRanges\\(\\[(?<" + SEQUENCE_POINTS_GROUP_NAME + ">" + SEQUENCE_POINT_PATTERN + "(," + SEQUENCE_POINT_PATTERN + ")*)\\]\\);\\s*+</script>.*",
```

Reply:
I fixed a code smell by not escaping it. and the unit tests pass.

could you help with a test case to cover adding it back?

Reply:
Hmm, it shouldn't be there indeed. I've learned something new 🙂